### PR TITLE
Declare dropped support for IE7

### PR DESCRIPTION
**Pull request by mightyiam**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ WYMeditor.
     but it is recommended that existing custom dialogs be rewritten using the
     new `wym.dialog` API (see API documentation) or at least that they be
     tested.
+  * Support for IE7 dropped.
```
Comment (winhamwr) [inline CHANGELOG.md]: Indentation is off

**Comment by winhamwr** [comment]
```shell
$ git grep ie7
dist/wymeditor/plugins/structured_headings/jquery.wymeditor.structured_headings.js:                         'structured_headings_ie7_editor.css';
docs/wymeditor_development/future_planning/seamless_ui.rst:`SO Answer <http://stackoverflow.com/questions/4455541/how-to-remove-iframe-border-in-ie7>`_.
docs/wymeditor_plugins/included_plugins/structured_headings.rst:   from its stylesheet ``structured_headings_ie7_user.css`` available in the
src/test/unit/plugin_tests/structured_headings.js:        expectedFileName = 'structured_headings_ie7_editor.css';
src/test/unit/plugin_tests/structured_headings.js:        stylesheetURL += 'structured_headings_ie7_user.css';
src/test/unit/plugin_tests/structured_headings.js:    module("structured_headings-ie7_polyfill", {setup: prepareUnitTestModule});
src/test/unit/specific_feature_tests/exec_toggles.js:                msie7: "strong",
src/test/unit/specific_feature_tests/exec_toggles.js:                msie7: "em",
src/wymeditor/plugins/structured_headings/jquery.wymeditor.structured_headings.js:                         'structured_headings_ie7_editor.css';
```

```shell
$ git grep -i "internet explorer 7"
README.rst:May also be of interest, is that WYMeditor fully supports Internet Explorer 7.
dist/website/index.html:<p>May also be of interest, is that WYMeditor fully supports Internet Explorer 7.</p>
docs/wymeditor_development/testing_wymeditor.rst:  In Internet Explorer 7, after the first test run, it appears that subsequent
```

Looks like we have some IE7 things that can be dropped and some further documentation that needs updating.

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,8 @@ WYMeditor.
     but it is recommended that existing custom dialogs be rewritten using the
     new `wym.dialog` API (see API documentation) or at least that they be
     tested.
+* [#679](https://github.com/wymeditor/wymeditor/pull/679)
+  Support for IE7 dropped.
```
Comment (mightyiam) [inline CHANGELOG.md]: Fixed and added link.

File: docs/wymeditor_plugins/included_plugins/structured_headings.rst [hunk 3]
```diff
@@ -35,32 +33,8 @@ these steps:
...
-   assigned numbering in WYMeditor and adds the numbering for these headings
-   into the page. This function should be called in a script on document ready
-   after the ``jquery.wymeditor.structured_headings.js`` script is included on
-   the page. Here is an example of what that script should look like:
-
-    .. code-block:: javascript
-
-        jQuery(document).ready(function() { numberHeadingsIE7(); });
-
-Browser Support
```
Comment (mightyiam) [inline docs/wymeditor_plugins/included_plugins/structured_headings.rst]: Decided to drop this section entirely.

File: src/test/unit/plugin_tests/structured_headings.js [hunk 4]
```diff
@@ -5,7 +5,6 @@
     QUnit,
```
Comment (mightyiam) [inline src/test/unit/plugin_tests/structured_headings.js]: Can we trust the test coverage for this feature?

File: src/test/unit/utils.js [hunk 5]
```diff
@@ -46,17 +46,13 @@ var preLt = /</g;
...
-    'title',
-    'target'
+var disposableAttributes = [
+    '_wym_visited',
+    // IE8
+    'nodeindex',
+    // IE8 uses this for img elements
+    'complete',
+    '_moz_editor_bogus_node'
 ];
```
Comment (mightyiam) [inline src/test/unit/utils.js]: Logic is flipped.

File: src/test/unit/utils.js [hunk 6]
```diff
@@ -267,28 +263,25 @@ function normalizeHtml(node) {
                 attr = attrs[i];
                 attrName = attr.nodeName.toLowerCase();
                 attrValue = attr.value;
-                keepAttr = false;
+                keepAttr = true;
```
Comment (mightyiam) [inline src/test/unit/utils.js]: Logic is flipped.

File: src/wymeditor/editor/base.js [hunk 7]
```diff
@@ -974,49 +974,19 @@ WYMeditor.editor.prototype.isListNode = function (node) {
...
-
-   Unlike sane browsers, IE7 provides all possible attributes in a node's
-   `attributes`. Thus, a simple check like `attributes.length > 0` isn't
-   enough (as long as we support IE7). This function tries to determine
-   whether the element really does have meaningful attributes, considering
-   IE7's behavior. This turns out to be not so simple, as IE7 populates
-   some attributes with truth-y values. So the mechanism is a kind of
-   a blacklist filter of IE7's junk-attributes. If an attribute passes this
-   blacklist filter, then this function returns true.
+   necessary.
```
Comment (mightyiam) [inline src/wymeditor/editor/base.js]: I remember writing this.

File: src/wymeditor/editor/base.js [hunk 8]
```diff
@@ -974,49 +974,19 @@ WYMeditor.editor.prototype.isListNode = function (node) {
...
-        // Any attribute with these values isn't interesting
-        falsyAttrValues = [
-            '',
-            undefined,
-            false,
-            null,
-            'null'
+            'length',
+            'ie8_length'
         ];
```
Comment (mightyiam) [inline src/wymeditor/editor/base.js]: IE8 is much nicer than 7.

**Comment by mightyiam** [comment]
The seamless skin has some IE7 specific things. I'm not sure about anything that has to do with skins. Never before touched that aspect of WYMeditor.

I don't even know if the seamless skin is the default skin or whether it is used by anyone or whether it even works.

@winhamwr advice?

File: src/test/unit/plugin_tests/structured_headings.js (hunk 4 above)
Comment (winhamwr) [inline src/test/unit/plugin_tests/structured_headings.js]: I think it's pretty good, yes.

**Comment by winhamwr** [comment]
> The seamless skin has some IE7 specific things.

I'm cool with leaving those in for now.

File: README.rst [hunk 9]
```diff
@@ -99,7 +99,7 @@ required in your project, and you want to ensure that users are focused on the
 structure of their content instead of tweaking fonts and margins, perhaps you
 should give WYMeditor a try.
 
-May also be of interest, is that WYMeditor fully supports Internet Explorer 7.
+May also be of interest, is that WYMeditor fully supports Internet Explorer 8.
```
Comment (winhamwr) [inline README.rst]: Weird wording here. How about:

WYMeditor also fully supports Internet Explorer 8.

File: src/test/unit/plugin_tests/structured_headings.js [hunk 10]
```diff
@@ -15,32 +14,6 @@
 /*
     Tests for the structured_headings plugin.
 */
-var START_NODE_CLASS = WYMeditor.STRUCTURED_HEADINGS_START_NODE_CLASS;
```
Comment (winhamwr) [inline src/test/unit/plugin_tests/structured_headings.js]: Hooray for being to rip out all of this code!

**Comment by winhamwr** [comment]
Looks good!

**Comment by mightyiam** [comment]
Thank you!

File: README.rst (hunk 9 above)
Comment (mightyiam) [inline README.rst]: Fixed in master.

**Comment by winhamwr** [comment]
There was a jshint error in this. [failing build](https://travis-ci.org/wymeditor/wymeditor/builds/53213039).

Still not sure why travis stopped posting build statuses to our pull requests :(

**Comment by mightyiam** [comment]
Oops. Fixed.

**Pull request merged**